spec/queryable_spec.cr

```diff
@@ -1471,4 +1484,22 @@ describe Avram::Queryable do
       query.to_sql.should eq original_query_sql
     end
   end
+
+  describe "with query cache" do
+    it "only runs the query once" do
+      # We're testing the actual caching
+      Fiber.current.query_cache = LuckyCache::MemoryStore.new
+      Avram.temp_config(query_cache_enabled: true) do
+        UserFactory.create &.name("Amy")
+        UserQuery.query_counter.should eq(0)
+
+        UserQuery.new.name("Amy").first
+        UserQuery.new.name("Amy").first
+        user = UserQuery.new.name("Amy").first
```

Comment:
What happens if you update some other column on users and then fetch in the same way?

Reply:
You'd get the cached version, not the updated version. Though, inserts and updates aren't cached here, which means if you did update a record, you'd have access to that updated record through the save operation. As of right now, the other option would be to flush all the cache.

We could do a `cache.delete(the_key)` type deal, but that goes to my question about the key. You'd have to know how it was derived... We can't just put a `cache_key` method on the models, because you're not necessarily caching a single model. Maybe `cache_key` becomes a public method?